lib/puppet/provider/service/redhat.rb

```diff
@@ -39,6 +41,7 @@ def enabled?
   # Don't support them specifying runlevels; always use the runlevels
   # in the init scripts.
   def enable
+      chkconfig("--add", @resource[:name])
```

Comment:
@joshcooper This has been refactored to use the netconsole service,and to force provider => redhat which was needed to ensure the redhat provider was used on RHEL v7. I think this should take care of the issues you've raised.